addons/website/static/src/snippets/s_share/000.js

```diff
@@ -31,47 +20,67 @@ const ShareWidget = publicWidget.Widget.extend({
-                return a + url + c;
-            })
-            .replace(this.TITLE_REGEX, function (match, a, b, c) {
-                if (aEl.classList.contains('s_share_whatsapp')) {
-                    // WhatsApp does not support the "url" GET parameter.
-                    // Instead we need to include the url within the passed "text"
-                    // parameter, merging everything together, e.g of output:
-                    // https://wa.me/?text=%20OpenWood%20Collection%20Online%20Reveal%20%7C%20My%20Website%20http%3A%2F%2Flocalhost%3A8888%2Fevent%2Fopenwood-collection-online-reveal-2021-06-21-2021-06-23-8%2Fregister
-                    // For more details, see https://faq.whatsapp.com/general/chats/how-to-use-click-to-chat/
-                    return `${a + title}%20${url + c}`;
-                }
-                return a + title + c;
-            })
-            .replace(this.MEDIA_REGEX, (match, a, b, c) => {
-                return a + media + c;
-            });
+        const urlParamsFound = urlParams.find(param => modifiedUrl.searchParams.has(param));
+        if (urlParamsFound) {
+            modifiedUrl.searchParams.set(urlParamsFound, currentUrl);
+        }
+
+        const titleParamsFound = titleParams.find(param => modifiedUrl.searchParams.has(param));
+        if (titleParamsFound) {
+            if (aEl.classList.contains('s_share_whatsapp')) {
+                // WhatsApp does not support the "url" GET parameter.
+                // Instead we need to include the url within the passed "text"
+                // parameter, merging everything together, e.g of output:
+                // https://wa.me/?text=%20OpenWood%20Collection%20Online%20Reveal%20%7C%20My%20Website%20http%3A%2F%2Flocalhost%3A8888%2Fevent%2Fopenwood-collection-online-reveal-2021-06-21-2021-06-23-8%2Fregister
+                // For more details, see https://faq.whatsapp.com/general/chats/how-to-use-click-to-chat/
+                modifiedUrl.searchParams.set(titleParamsFound, `${currentTitle}%20${currentUrl}`);
+            } else {
+                modifiedUrl.searchParams.set(titleParamsFound, currentTitle);
+            }
+        }
+
+        const mediaParamsFound = mediaParams.find(param => modifiedUrl.searchParams.has(param));
+        if (mediaParamsFound) {
+            const ogImageEl = document.querySelector("meta[property='og:image']");
+            // Some pages (/profile/user/ID) don't have an image to share.
+            if (ogImageEl) {
+                const media = encodeURIComponent(ogImageEl.content);
+                modifiedUrl.searchParams.set("media", media);
+            } else {
+                modifiedUrl.searchParams.delete("media");
+            }
+        }
 
-        window.open(aEl.href, aEl.target, 'menubar=no,toolbar=no,resizable=yes,scrollbars=yes,height=550,width=600');
+        window.open(
+            decodeURIComponent(modifiedUrl.toString()),
```

Comment:
Does this need to be decoded? :thinking: Why would we encode it in the first place if it is to decode it afterwards?

Reply:
Indeed no. We don't have to handle the encode/decode stuff ourself now as we use `searchParams`.

This and the other points are now fixed. It's ready for you review.